src/sqlancer/ReducerExample.java

```diff
@@ -0,0 +1,120 @@
+package sqlancer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import sqlancer.FoundBugException.Reproducer;
+import sqlancer.common.query.Query;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.sqlite3.SQLite3GlobalState;
+
+public class ReducerExample<G extends GlobalState<O, ?, C>,
+        O extends DBMSSpecificOptions<?>, C extends SQLancerDBConnection>
+        implements Reducer <G, O, C> {
+    private final DatabaseProvider<G, O, C> provider;
+    private boolean observedChange;
+
+    private static final String[] TOKENS = new String[] {
+        "OR IGNORE", "OR ABORT", "OR ROLLBACK", "OR FAIL", "TEMP", "TEMPORARY",
+        "UNIQUE", "NOT NULL", "COLLATE BINARY", "COLLATE NOCASE",
+        "COLLATE RTRIM", "INT", "REAL", "TEXT", "IF NOT EXISTS", "UNINDEXED"};
+
+    public ReducerExample(DatabaseProvider<G, O, C> provider) {
+        this.provider = provider;
+    }
+
+    @SuppressWarnings("unchecked")
+    public void reduce(G state, Reproducer reproducer, G newGlobalState) throws Exception {
+        
+        List<Query<C>> knownToReproduceBugStatements = new ArrayList<Query<C>>();
+        for (Query<?> stat : state.getState().getStatements()) {
+            knownToReproduceBugStatements.add((Query<C>) stat);
+        }
+        System.out.println("Starting query:");
+        printQueries(knownToReproduceBugStatements);
+        System.out.println();
+
+        do {
+            observedChange = false;
+            knownToReproduceBugStatements = tryReduction(state, reproducer, newGlobalState,
+            knownToReproduceBugStatements, (candidateStatements, i) -> {
+                candidateStatements.remove((int) i);
+                return true;
+            });
+        } while (observedChange);
+
+        for (String s : TOKENS) {
```

Comment:
For an initial implementation, I'd suggest keeping it general and simple, by only removing complete statements. This might simplify the implementation.

Reply:
Done.